### ISSUE TITLE
Android SDK. There's already another call in progress. Please end it first and try again.

### ISSUE THREAD

**Issue by Jakebreen** [jitsi/jitsi-meet#4720]
Hello,

Android SDK 19.4.0.
I have found that when my application launches the Activity that extends **JitsiMeetActivity** JitsiMeet works well with both the Jitsi server endpoint _(https://meet.jit.si)_ and my own server endpoint. This setup uses the default Jitsi UI like the welcomepage.

However, if i launch this activity from another custom activity (my own welcome page), the first join attempt is successful but every subsequent join attempt after the first successful join fails with a dialog displaying that says:

> "There's already another call in progress. Please end it first and try again."

This doesn't make sense as to return to the previous activity that the join button exists on, I mist first have had to **End** the call.

What is so different between the Apps default activity launcher being the activity that extends from JitsiMeetActivity as opposed to launching that activity through an intent?

**Comment by saghul** [comment]
Indeed, having a dangling connection will trigger that behavior. How are you handing up the call to return to your activity?

Can you share the logcat output when you do that?

**Comment by Jakebreen** [comment]
I'm using the red hang-up button that is present in the native Jitsi in-call screen, it returns me to my previous activity. ConferenceOptions WelcomePage is disabled.

Log uploaded:
https://gist.github.com/Jakebreen/e5a0911df2003cb53fe2f59198c6e675

**Comment by saghul** [comment]
How are you setting the global and conference specific options? Looks like you are joining the conference again after hangup, which is probablhy the source of the problem.

**Comment by Jakebreen** [comment]
```
@Override
    protected void initialize() {
        Log.d("XXX", "initialize()");
        final Intent intent = getIntent();
        if (intent == null) return;

        final String room = intent.getStringExtra("room");
        final String username = intent.getStringExtra("username");

        JitsiMeetConferenceOptions defaultOptions
                = new JitsiMeetConferenceOptions.Builder()
                .setWelcomePageEnabled(false)
                .setServerURL(buildURL(intent.getStringExtra("url")))
                .setRoom(room)
                .setUserInfo(getUserInfo(username))
                .build();
        JitsiMeet.setDefaultConferenceOptions(defaultOptions);

        super.initialize();
    }
```


The overridden initialize method of the JitsiMeetActivity extended class. This method is not called a second time after hang-up. Do i need to invalidate the conference options when the call is terminated?

**Comment by saghul** [comment]
There you go :-)

The default options apply to all conferences, adn are re-applied after one ends. Because you are setting a room there, it will re-join it. I'll try to come up with something to make this more obvious, or throw an exception.

You must create a new object which you pass to `join`, and call setRoom on that one.

Same goes for `setUserInfo`. Just set the server on the default ones.

**Comment by Jakebreen** [comment]
Thanks for help.

**Comment by piotrkst** [comment]
I do have this problem occurring in my project, but the configurations are set properly. Once it starts occuring, it won't go away. I have some reports that people's phones are restarting after some time :O

Some logs I've found important:

E/JitsiMeetSDK: [modules/statistics/AnalyticsAdapter.js] Required field missing (containerId, containerType, objectId or objectType)
E/JitsiMeetSDK: [modules/statistics/AnalyticsAdapter.js] Dropping a mis-formatted event: {"type":"track","action":"unpinned","actionSubject":"participant","objectType":"participant","attributes":{"participant_count":1},"source":"unpinned","containerType":"conference"}
I/JitsiMeetSDK: [modules/xmpp/strophe.ping.js] Ping interval cleared
D/JitsiMeetSDK: [features/base/lastn] There is no active conference, not updating last N
I/JitsiMeetSDK: [features/base/connection] No connection found while disconnecting.
W/JitsiMeetSDK: [features/base/connection] JitsiConference.leave() rejected with: Error(Error){"message":"The timeout for the confirmation about leaving the room expired.",(...)

**Comment by afsaredrisy** [comment]
Facing the same issue if I am killing the app during a meeting (Swiping up app from task manager) & then if again I am trying to create a new meeting its showing call already exists. I am not even able to use the phone's call feature until I restart the device.  Here is how I am using JitsiMeet.

```java
view = new JitsiMeetView(this);
JitsiMeetUserInfo jitsiMeetUserInfo = new JitsiMeetUserInfo();
jitsiMeetUserInfo.setDisplayName(userBean.getName());
jitsiMeetUserInfo.setDisplayName(displayname);
   JitsiMeetConferenceOptions options = null;
        try {
            options = new JitsiMeetConferenceOptions.Builder()
                    .setServerURL(new URL("https://meet.jit.si"))
                    .setRoom(AppConstants.MEETING_ID)
                    .setUserInfo(jitsiMeetUserInfo)
                    .setFeatureFlag("invite.enabled", false)
                    .setFeatureFlag("pip.enabled", true)
                    .setWelcomePageEnabled(false)
                    .build();
        } catch (MalformedURLException e) {
            e.printStackTrace();
        }
view.join(options);

```
Is there anything I can do to resolve this error?

**Comment by saghul** [comment]
Set the `call-integration.enabled` flag to false.

But more importantly, since  you are using the view API, you should call leave() when your activity is going to be destroyed too.

**Comment by akshatakki2000** [comment]
Can you explain more as to how to call leave() is it a predefined function? if not what all to add?
